### PR TITLE
First few steps of refactoring website to Typescript

### PR DESCRIPTION
**Pull request by minkimcello**
## TODOs and Qs
- The width properties are being prettier-fied. I'll see how I can configure prettier to format the fractions. this started happening after i installed prettier a dev dependency, that's why it's happening, right?

- `tsconfig.json` i copied most of the config settings of `@resideo/blueprint` and looked up each setting in the typescript docs. They all sort of make sense but not having put typescript to real use before, can you double check the configurations?

- Unlike other typescript projects, gatsby's plugin for typescript seems to mitigate the need to have a tsc build script? I'm guessing it builds internally automatically? because when i run `gatsby develop` it knows which files to build without me having to build manually and somehow point the website to the build directory.

- [`provider.tsx`](https://github.com/bigtestjs/bigtest/blob/mk/typescripter/packages/website/src/theme/provider.tsx): would the `ThemeProvider` also be `React.FC`?

- [`global.ts`](https://github.com/bigtestjs/bigtest/blob/mk/typescripter/packages/website/src/theme/global.ts): What's the difference between typing the color property as string vs `CSS.ColorProperty`? Is it just a matter of being more specific?
  - can you help me understand the syntax of `<{ theme: Theme }>` for `createGlobalStyle`?
    - in [`404.tsx`](https://github.com/bigtestjs/bigtest/blob/mk/typescripter/packages/website/src/pages/404.tsx) i typed the component as `React.FC` but in the blog post mentioned below, he types `React.RC<StyledSystemProps>`.
  - should [`base.ts`](https://github.com/bigtestjs/bigtest/blob/mk/typescripter/packages/website/src/theme/base.ts) be a lot more of `CSS.*` types?

- for styled-system types, i imported `@types/styled-system`. is the standard practice to look at the available types from [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/styled-system/index.d.ts)? how would i import it?
  - from [this](https://dev.to/jdcas89/start-your-app-the-right-way-featuring-react-styled-system-styled-components-and-typescript-7a4) blog on using typescript with styled-system/styled-components, he exports type `StyledSystemProps`. Could you show me how I should refactor `box.js` using those styled system props?

### CODE REVIEW

File: packages/website/src/theme/base.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import { Theme } from 'styled-system';
+
+export interface LineHeights {
+	body: number;
```
Comment (jorgelainfiesta) [inline packages/website/src/theme/base.ts]: The spacing here seems to be off?

Comment (jorgelainfiesta) [inline packages/website/src/theme/base.ts:4]: The spacing in this page seem to be off?

**Review by jorgelainfiesta — COMMENTED** [review]
1. The linting of the width fraction's alright. Having Prettier as a dependency too, it'll help us have consistency across devs. 

2. Looks good! (also you should refer to other sources, that one is a private project).

3. `gatsby-plugin-typescript` takes care of pretty much everything.

4. Well it is a component so I don't see why not, but I'm not sure.

5.1 Indeed, it's about specificity. With `string` it will accept anything, but with `CSS.ColorProperty` it'l check if it's a valid CSS color.

5.2 Those string literals used by `styled-components` will have the theme available by default. I'm not sure why you have to describe it explicitly in this case. 

5.3 You would need to use `React.RC<StyledSystemProps>` if your component is expecting any of the styled system props (padding, margin, width, etc) from whoever consumes it. The 404 page shouldn't be a case of that.

6.1 Type definitions in files like `**.d.ts` are surveyed automatically by Typescript so there's no need to import them explicitly on every file you use them. 

6.2 The box definition only puts together APIs from `styled-components` and `styled-system` that are already typed, so I think we don't need to do any explicit typing ourselves.

File: packages/website/src/theme/base.ts (hunk 1 above)
Comment (minkimcello) [reply packages/website/src/theme/base.ts:4]: I'm not sure why it's being displayed like that. It's spaced/tabbed properly locally.

**Comment by cowboyd** [comment]
Can we move this over to https://github.com/thefrontside/bigtest ?

**Comment by minkimcello** [comment]
@cowboyd will do

**Comment by minkimcello** [comment]
Moved [here](https://github.com/thefrontside/bigtest/pull/108).

**Pull request closed without merging**